api/src/main/java/io/opentelemetry/tags/unsafe/TagMapInScope.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.tags.unsafe;
+
+import io.grpc.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.tags.TagMap;
+
+/**
+ * A scope that manages the {@link Context} for a {@link TagMap}.
+ *
+ * @since 0.1.0
+ */
+public final class TagMapInScope implements Scope {
+  private final Context orig;
+
+  private TagMapInScope(TagMap tags) {
+    orig = ContextUtils.withValue(tags).attach();
+  }
+
+  /**
+   * Constructs a new {@link TagMapInScope}.
+   *
+   * @param tags the {@code TagMap} to be added to the current {@code Context}.
+   * @since 0.1.0
+   */
+  public static TagMapInScope create(TagMap tags) {
+    return new TagMapInScope(tags);
+  }
+
+  @Override
+  public void close() {
+    Context.current().detach(orig);
```

Comment:
Based on the `Context` documentation, we shouldn't do this, but save the context variable:

> If this method finds that this context is not current, either you or some code in-between are not detaching correctly, and a SEVERE message will be logged but the context to attach will still be bound. Never use Context.current().detach(), as this will compromise this error-detecting mechanism.
```

But maybe we should do that in another PR...

Reply:
Not sure I understand the comment. Can someone clarify?

Reply:
The comment says we should not be using `Context.curret()` but rather store the current into the private field in `Scope`impl when the scope is created and at the close time call `current.detach(orig)`

Reply:
Let's create an issue and discuss about this, I will add the gRPC devs there as well.

Reply:
Let's just open a (minor) Issue for this, so we don't get blocked. We can later update the code.